File: 3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.py
class Solution:
    def minimumCost(self, nums: List[int]) -> int:
        min1 = float('inf') #setting it to infinity so it remains as larger number 
        min2 = float('inf')
        for i in nums[1:] :
            if i < min1 :
                min2 = min1 
                min1 = i
            elif i < min2 :
                min2 = i
        return nums[0] + min1 + min2

        

File: 3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.java
class Solution {
    public int minimumCost(int[] nums) {
        int min1 = Integer.MAX_VALUE;
        int min2 = Integer.MAX_VALUE;
        //give the min1 and min 2 with the largest values like 2147483647 something
        for(int i =1 ; i<nums.length;i++){
            if(nums[i]<min1){
                min2=min1;
                min1 = nums[i];
            }
            else if (nums[i]<min2){
                min2 = nums[i];
            }

        }
        return nums[0]+min1 + min2;
    }
}